### ISSUE TITLE
Blocking a direct connection when it is not a direct connection

### ISSUE THREAD

**Issue by DevilInMachine** [safing/portmaster#345]
**What happened**:
About once every hour or several hours, I can't open any sites in the browsers. The browser logs in Portmaster at this moment show only the IP address for the site I'm trying to open, but not the domain.

**What did you expect to happen?**:

I expect that all browser connections will be correctly mapped to DNS queries and will not be displayed as direct connection attempts.

**How did you reproduce it?**:

I'm really not sure, but I have activated the functions: Ignore System/Network Servers; Enforce Secure DNS; Beta Privacy Filter; Block P2P/Direct Connections; Beta Block Bypassing; Beta Disable Auto Allow

Maybe someone else can give more specific information? An interesting point is that the problem stops if you briefly disable filters, although I absolutely do not understand the connection between filters and DNS/direct connections.

**Debug Information**:

I will send it by mail with a link to this bug report, so as not to worry about deleting some private data. Just delete the files when you view them :)

**Comment by ppacher** [comment]
Hi @DevilInMachine, thanks for reporting.

This seems like an issue with the DNS cache of either Portmaster or your browser. Could you try to open a website that's not in the browsers cache the next time you experience this issue? This will help to identify the actual issue. Just open something you haven't visited for some time.

In the meanwhile you can switch to the [Beta Release Channel](https://docs.safing.io/portmaster/settings#core/releaseChannel). We have re-written and improved the DNS resolver in Portmaster and those changes are already released to the beta channel. This might fix your issue as well.

**Comment by DevilInMachine** [comment]
I tried to open a site that I haven't visited for a very very long time, and also cleared the cache, but nothing worked and it's very strange. I still don't understand anything, but I can say that the problem really doesn't repeat itself in the current beta (or did it help that I changed quad9 to a manually configured controld?). Perhaps the bug can be closed, but I will let you know if the problem persists.
